packages/core-ethereum/src/indexer/index.ts

```diff
@@ -833,7 +847,11 @@ class Indexer extends (EventEmitter as new () => IndexerEventEmitter) {
     log('New node announced', account.get_address().to_hex(), account.get_multiaddress_str())
     metric_numAnnouncements.increment()
 
-    await this.db.updateAccountAndSnapshot(account, lastSnapshot)
+    assert(lastSnapshot !== undefined)
```

Comment:
This will crash the node, is that the desired behaviour? Can it recover upon restart from it?

Reply:
It doesnt